app/static/tests/e2e/code.etest.ts

```diff
@@ -86,11 +87,19 @@ test.describe("Code Tab tests", () => {
     });
 
     test("can reset code editor", async ({ page }) => {
-        const invalidCode = "deriv(y1) test * faker";
+        const invalidCode = "faker\n";
         await writeCode(page, invalidCode);
+        await expect(await page.locator(".run-tab .run-update-msg")).toHaveText(
+            "Model code has been updated. Compile code and Run Model to view updated graph.", {
+                timeout
+            }
+        );
         await expect(await page.innerText(".wodin-left .wodin-content #reset-btn")).toBe("Reset");
         await expect(await page.innerText(".wodin-left .wodin-content")).toContain("faker");
+        await expect(await page.innerText(".wodin-left .wodin-content #code-status")).toContain("Code is not valid");
         await page.click("#reset-btn");
+        await page.waitForResponse((response) => response.url().includes("/odin"));
```

Comment:
I don't think you should need this..?

Reply:
Right, yes -  it's equivalent to the `await ...locator...toHaveText`s that we're using elsewhere. Wouldn't have thought of doing it that way!